### ISSUE TITLE
Pointer-to-member-function should reject explicit object parameter ("deducing this")

### ISSUE THREAD

**Issue by Quuxplusone** [Quuxplusone/llvm-project#24]
("Deducing this" implementation progress is being tracked in https://github.com/llvm/llvm-project/issues/59619 )

Clang accepts the following invalid code: https://godbolt.org/z/rr1TKejPz
```
struct S {
    int f(int);
};
using T = int (S::*)(this int);  // INVALID
T t = &S::f;
```

GCC correctly rejects:
```
error: a pointer to member function type cannot have an explicit object parameter
```

**Comment by Quuxplusone** [comment]
Fixed in f7c2e5fa05d221a3dfc53744f353517407c2ffec! 🎉

**Comment by Quuxplusone** [comment]
Probably will be resolved by https://github.com/llvm/llvm-project/pull/89078

**Comment by Sirraide** [comment]
That’s pretty much exactly the same code that was also in the other issue, so yeah, that should be fixed by that pr as well.

**Comment by Quuxplusone** [comment]
Fixed by #89078.
